devel/openssl3/Portfile

```diff
@@ -58,6 +58,13 @@ if {${os.platform} eq "darwin" && ${os.major} < 11} {
 #   To be seen if needed with 3.0.0, and if dodes needs updating.
 #patchfiles-append   patch-pre-Sierra.diff
 
+# https://github.com/openssl/openssl/pull/16584
+# https://github.com/openssl/openssl/issues/16551
+# Fixes "Undefined symbols for architecture i386: ___atomic_is_lock_free"
+if { ${configure.build_arch} eq "i386" } || "i386" in ${configure.universal_archs} {
```

Comment:
Note that this patch will apply on x86_64 systems even when the universal variant is not requested, so long as i386 is in universal_archs. That would warrant a revision increase since it changes things for people who already had openssl3 x86_64 installed.

Reply:
I’d recent done a similar change for librsvg and the fallback wasn’t triggered when not compiling as +universal (cargo/rust got added to deps)

When compiling as +universal the fallback was forced.

Reply:
I can't speak to that, but what I stated remains true.

Reply:
It's easy enough to verify by putting a `ui_msg "patch applied"` inside that conditional:
```
$ port info --name openssl3 build_arch=x86_64 -universal
patch applied
name: openssl3
$ port info --name openssl3 build_arch=x86_64 +universal
patch applied
name: openssl3
$ port info --name openssl3 build_arch=i386 -universal
patch applied
name: openssl3
$ port info --name openssl3 build_arch=i386 +universal
patch applied
name: openssl3
```

Reply:
Not near my system but what happens if you change the first part of the if statement to `${build_arch} eq "i386"`

Reply:
Or more what would be the correct way to handle this so it’s only getting applied for i386 or a universal build that contains i386?

Reply:
I would not expect that to change anything.

If you want to restrict something to when the universal variant is set, you have to add the condition `${universal_possible} && [variant_isset universal]`.

```
if {(${configure.build_arch} eq "i386") || (${universal_possible} && [variant_isset universal] && "i386" in ${configure.universal_archs})} {
```

Since some users may already have built the previous version, the revision should be increased when this change is made.

Reply:
So ether this will need to wait on me getting home and opening a PR or a Macports maintainer to push the the change and bump the revisit.